src/omv/ports/mimxrt/sensor.c

```diff
@@ -181,13 +184,74 @@ void sensor_sof_callback() {
     vbuffer_t *buffer = framebuffer_get_tail(FB_PEEK);
     if (buffer == NULL) {
         sensor_abort(false, true);
-    } else if (buffer->offset < resolution[sensor.framesize][1]) {
+        return;
+    } else if ((sensor.pixformat != PIXFORMAT_JPEG) && (!sensor.transpose) && (!sensor_get_cropped())) {
+        // Drop the frame here to avoid wasting SDRAM bandwidth.
+        sensor_throttle_framerate();
+        if (sensor.drop_frame) {
+            return;
+        }
+        CSI_REG_DMASA_FB1(CSI) = (uint32_t) buffer->data;
+        CSI_REG_DMASA_FB2(CSI) = (uint32_t) buffer->data;
+        CSI_REG_IMAG_PARA(CSI) =
+            ((MAIN_FB()->u * sensor_get_src_bpp()) << CSI_IMAG_PARA_IMAGE_WIDTH_SHIFT) |
+            (MAIN_FB()->v << CSI_IMAG_PARA_IMAGE_HEIGHT_SHIFT);
+    } else if ((sensor.pixformat != PIXFORMAT_JPEG) && (buffer->offset < resolution[sensor.framesize][1])) {
         // Missed a few lines, reset buffer state and continue.
         buffer->reset_state = true;
     }
+    // Clear the FIFO and re/enable DMA.
+    CSI_REG_CR3(CSI) |= (CSI_CR3_DMA_REFLASH_RFF_MASK | CSI_CR3_DMA_REQ_EN_RFF_MASK);
+}
+
+#if defined(OMV_CSI_DMA)
+int sensor_dma_memcpy(void *dma, void *dst, void *src, int bpp, bool transposed) {
+    // EMDA will not perform burst transfers for anything less than 32-byte chunks made of four 64-bit
+    // beats. Additionally, the CSI hardware lacks cropping so we cannot align the source address.
+    // Given this, performance will be lacking on cropped images. So much so that we we do not use
+    // the EDMA for anything less than 4-byte transfers...
```

Comment:
```suggestion
    // Given this, performance will be lacking on cropped images. So much so that we do not use
    // the EDMA for anything less than 4-byte transfers.
```

Reply:
Fixed the comment.